feat: add SCIM provisioning via Okta (#4132)
coder/coder: docs/admin/auth.md

Once complete, run `sudo service coder restart` to reboot Coder.

> When a new user is created, the `preferred_username` claim becomes the username. If this claim is empty, the email address will be stripped of the domain, and become the username (e.g. `example@coder.com` becomes `example`).

## SCIM

> Coder supports user provisioning and deprovisioning via SCIM 2.0 with header
> authentication. Upon deactivation, users are deactivated within the platform but
> are not deleted. Configure your SCIM application with an auth key and supply it
> to Coder.

Reviewer: ```suggestion
Coder supports user provisioning and deprovisioning via SCIM 2.0 with header
authentication. Upon deactivation, users are [suspended](userd.md#suspend-a-user)
[suspended](userd.md#suspend-a-user) are not deleted. [Configure](./configure.md)
your SCIM application with an auth key and supply it the Coder server.
```

Reviewer: just some wording/consistency changes, thanks for writing the docs!
